MTM-63159: Common Security Scenarios and Best Practices for Device Certificates Management (#3776)
SoftwareAG/c8y-docs: content/device-certificate-authentication/device-certificates.md

{{< c8y-admon-info >}}
If administrators cannot carry out this process on their own for organizational reasons, they can manually request the proof of possession for the corresponding certificate and the {{< company-c8y >}} Support team can complete the proof of possession through a back end API upon reasonable verification.
{{< /c8y-admon-info >}}

> ### Common Security Scenarios and Best Practices

Reviewer: ```suggestion
### Common security scenarios and best practices {#common-security-scenarios-and-best-practices}
```

> #### What to do when a device’s certificate is compromised?

Reviewer: ```suggestion
#### What to do when a device’s certificate is compromised? {#what-to-do-when-a-device-certificate-is-compromised}
```

If you suspect or confirm that a device certificate has been compromised, follow the steps below:

> 1. **Revoke the Compromised Certificate**

Reviewer: ```suggestion
1. **Revoke the compromised certificate**
```

>     - Add the certificate to the [Certificate Revocation List](/device-certificate-authentication/managing-trusted-certificate-settings/#crl-settings) maintained by {{< product-c8y-iot >}}.

Reviewer: ```suggestion
    - Add the certificate to the [certificate revocation list](/device-certificate-authentication/managing-trusted-certificate-settings/#crl-settings) maintained by {{< product-c8y-iot >}}.
```

> 2 **Disable the device user**

Reviewer: ```suggestion
2. **Disable the device user**
```

    - This will disconnect the device from the {{< product-c8y-iot >}} but leave the certificate untouched until an investigation can be done.

>     - If not compromised:** Re-enable the device user. The device will reconnect using its existing certificate.

Reviewer: ```suggestion
    - If not compromised: Re-enable the device user. The device will reconnect using its existing certificate.
```

>     - If compromised:** Revoke the certificate to prevent any further use of it.

Reviewer: ```suggestion
    - If compromised: Revoke the certificate to prevent any further use of it.
```

> 3 **Provision a New Certificate**

Reviewer: ```suggestion
3. **Provision a new certificate**
```

    - Generate a new certificate for the device.
    - Using the existing CA certificate which is owned by {{< product-c8y-iot >}} generate a new certificate for the device.
    - Update the device to use the new certificate for authentication.

> #### What to do when the enrollment process has been compromised?

Reviewer: ```suggestion
#### What to do when the enrollment process has been compromised? {#what-to-do-when-the-enrollment-process-has-been-compromised}
```

Enrollment processes may be compromised when insecure practices are used, for example:
- Using a device ID-derived one-time password (OTP)
- Using the same OTP across all enrollment requests

If such a situation occurs, proceed with the following steps:

1. **Disable auto-registration on the tenant’s CA certificate**
    - This stops any new automatic registrations immediately.
    - Notice that existing registered devices will remain connected and operational.
2. **Communicate and remediate unsafe procedures**
    - Inform all stakeholders about the issue.
    - If necessary, disable users who were following unsafe enrollment practices.
    - Provide guidance on secure manual procedures.
3. **Re-enable auto-registration**
    - Once safe practices have been confirmed and enforced, enable auto-registration again.
    - This allows new enrollment requests to be processed securely.

> > **Note:** Consider rotating enrollment credentials periodically and monitoring logs for unusual enrollment activity.

Reviewer: ```suggestion
{{< c8y-admon-info >}} 
Consider rotating enrollment credentials periodically and monitoring logs for unusual enrollment activity.
{{< /c8y-admon-info >}}
```

> #### What to do when a CA certificate has been compromised?

Reviewer: ```suggestion
#### What to do when a CA certificate has been compromised? {#what-to-do-when-a-ca-certificate-has-been-compromised}
```

If you suspect or confirm that a CA (Certificate Authority) certificate has been compromised, follow the steps below:

> 1. **Revoke the Compromised CA Certificate**

Reviewer: ```suggestion
1. **Revoke the compromised CA certificate**
```

> 1. **Revoke the Compromised CA Certificate**

Reviewer 2: minor: We can also mention that you can disable the certificate if CRL offline is not enabled.

>     - Add the compromised CA certificate to the [Certificate Revocation List](/device-certificate-authentication/managing-trusted-certificate-settings/#crl-settings) using the CRL API.

Reviewer: ```suggestion
    - Add the compromised CA certificate to the [certificate revocation list](/device-certificate-authentication/managing-trusted-certificate-settings/#crl-settings) using the CRL API.
```

    - Note: This action is effective only if the tenant has offline CRL check enabled in the {{< product-c8y-iot >}}.
    - The CA certificate itself remains in the system until the next step.

> 2. **Delete the Compromised CA**

Reviewer: ```suggestion
2. **Delete the compromised CA certificate**
```

    - Remove the compromised CA certificate from the tenant.
    - This action also deletes the associated key pair, ensuring it cannot be reused.

> 3. **Create a New CA**

Reviewer: ```suggestion
3. **Create a new CA certificate**
```

    - Generate a new CA certificate.
    - A new key pair will be created automatically during this process.
    - Use this new CA to issue device certificates going forward.

> 4. **Reprovision Devices**

Reviewer: ```suggestion
4. **Reprovision devices**
```

>     - All device certificates previously signed by the compromised CA will no longer be trusted by the {{< product-c8y-iot >}}.

Reviewer: ```suggestion
    - All device certificates previously signed by the compromised CA certificate will no longer be trusted by the {{< product-c8y-iot >}}.
```

    - Devices using these certificates will be unable to connect via any communication channel, such as MQTT or deviceAccessToken API.
    - Re-enroll affected devices using certificates issued by the new CA.

> > **Note:** After replacing a compromised CA, review and update your certificate management and enrollment policies to ensure stronger security practices, including periodic CA rotation and continuous monitoring for unusual certificate activity.

Reviewer: ```suggestion
{{< c8y-admon-info >}} 
After replacing a compromised CA certificate, review and update your certificate management and enrollment policies to ensure stronger security practices, including periodic CA rotation and continuous monitoring for unusual certificate activity.
{{< /c8y-admon-info >}} 
```

{{< c8y-admon-info >}}
If administrators cannot carry out this process on their own for organizational reasons, they can manually request the proof of possession for the corresponding certificate and the {{< company-c8y >}} Support team can complete the proof of possession through a back end API upon reasonable verification.
{{< /c8y-admon-info >}}

### Common Security Scenarios and Best Practices

#### What to do when a device’s certificate is compromised?

If you suspect or confirm that a device certificate has been compromised, follow the steps below:

1. **Revoke the Compromised Certificate**
    - Add the certificate to the [Certificate Revocation List](/device-certificate-authentication/managing-trusted-certificate-settings/#crl-settings) maintained by {{< product-c8y-iot >}}.

>     - The certificate itself will remain untouched until an investigation can be completed.

Reviewer 2: Currently, there is a logic that will remove this device from the user, and adding it to the CRL is an operation that cannot be undone.Once revoked, it will be permanently

Author: I will rephrase it then

> 2. **Prevent the compromised device from connecting to IoT services (MQTT broker, HTTPS**

Reviewer 2: As we talked before,  I don't think we can do it any other way than revoking the device certificate ourselves, or disable user.

    - deviceAccessToken API.
3. **Disable the device user**
    - This will disconnect the device from the {{< product-c8y-iot >}} but leave the certificate untouched until an investigation can be done.
    - If not compromised:** Re-enable the device user. The device will reconnect using its existing certificate.
    - If compromised:** Revoke the certificate to prevent any further use of it.
4. **Provision a New Certificate**
    - Generate a new certificate for the device.
    - Using the existing CA certificate which is owned by {{< product-c8y-iot >}} generate a new certificate for the device.
    - Update the device to use the new certificate for authentication.

#### What to do when the enrollment process has been compromised?

Enrollment processes may be compromised when insecure practices are used, for example:
- Using a device ID-derived one-time password (OTP)
- Using the same OTP across all enrollment requests

If such a situation occurs, proceed with the following steps:

1. **Disable auto-registration on the tenant’s CA certificate**

>     - This stops any new enrollment requests immediately.

Reviewer 2: Did we test it?

Author: I will test this scenario comprehensively
